asf_search/WKT/validate_wkt.py

```diff
@@ -0,0 +1,330 @@
+    output: merged-overlapping geometry
+    """
+    merge_report = None
+
+    if isinstance(geometry, BaseMultipartGeometry):
+        original_amount = len(geometry.geoms)
+        if original_amount == 1:
+            return geometry, merge_report
+
+        merged = unary_union(geometry)
+
+        # if there were non-overlapping shapes
+        if isinstance(merged, BaseMultipartGeometry):
+            unique_shapes = len(merged.geoms)
+            merged = orient(unary_union(GeometryCollection([geom.convex_hull for geom in merged.geoms])))
+            if isinstance(merged, BaseMultipartGeometry):
+                if unique_shapes != len(merged.geoms):
+                    merge_report = RepairEntry("'type': 'OVERLAP_MERGE'", f"'report': {unique_shapes - len(merged.geoms)} non-overlapping shapes merged by their convex-hulls")
+            else:
+                merge_report = RepairEntry("'type': 'OVERLAP_MERGE'", f"'report': {unique_shapes} non-overlapping shapes merged by their convex-hulls")
+        else:
+            merge_report = RepairEntry("'type': 'OVERLAP_MERGE'", f"'report': Overlapping {original_amount} shapes merged into one")
+
+        return merged, merge_report
+
+    return geometry, merge_report
+
+
+def _counter_clockwise_reorientation(geometry: Union[Point, LineString, Polygon]):
+    """
+        param geometry: Shapely geometry to re-orient
+        Ensures the geometry coordinates are wound counter-clockwise
+        output: counter-clockwise oriented geometry
+    """
+    reoriented_report = RepairEntry("'type': 'REVERSE'", "'report': Reversed polygon winding order")
+    reoriented = orient(geometry)
+    
+    if isinstance(geometry, Polygon):
+        # if the vertice ordering has changed
+        if reoriented.exterior.is_ccw != geometry.exterior.is_ccw:
+            return reoriented, reoriented_report
+
+    return reoriented, None
+
+
+def _get_clamped_geometry(shape: BaseGeometry) -> Tuple[BaseGeometry, List[RepairEntry]]:
+    """
+    param geometry: Shapely geometry to clamp    
+    Clamps geometry to +/-90 latitude and wraps longitude +/-180
+    output: clamped shapely geometry
+    """
+    coords_clamped = 0
+    coords_wrapped = 0
+    def _clamp_lat(x, y, z=None):
+        clamped = _clamp(y)
+        wrapped = x
+
+        if clamped != y:
+            nonlocal coords_clamped
+            coords_clamped += 1
+
+        return tuple([wrapped, clamped])
+
+    def _wrap_lon(x, y, z=None):
+        wrapped = x
+        if abs(x) > 180:
+            wrapped = (x + 180) % 360 - 180
+
+        if wrapped != x:
+            nonlocal coords_wrapped
+            coords_wrapped += 1
+
+        return tuple([wrapped, y])
+
+    def  _unwrap_lon(x, y, z=None):
+        unwrapped = x if x >= 0 else x + 360
+
+        return tuple([unwrapped, y])
+
+    wrapped = transform(_wrap_lon, shape)
+    unwrapped = wrapped
+    
+    if wrapped.bounds[2] - wrapped.bounds[0] > 180:
```

Comment:
wkt `POINT EMPTY` throws on this line

Reply:
Oh my comment didn't save! In the line where you loop through the geometries, you're still appending the geometry when it's empty, instead of a empty list. If you fix that, this should go away

Reply:
I went ahead and added a check for if the geometry is empty in the `validate_wkt()` method (it can even tell if a collection is empty). The part that recursively flattens geometry collections should work now since it's guaranteed not to get completely empty geometry now.